src/app/admin/packingSlotsTable/PackingSlotsTable.tsx

```diff
@@ -0,0 +1,332 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import supabase from "@/supabaseClient";
+import {
+    DataGrid,
+    GridActionsCellItem,
+    GridColDef,
+    GridEventListener,
+    GridRowEditStopReasons,
+    GridRowId,
+    GridRowModes,
+    GridRowModesModel,
+    GridRowsProp,
+    GridToolbarContainer,
+} from "@mui/x-data-grid";
+import Button from "@mui/material/Button";
+import AddIcon from "@mui/icons-material/Add";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/DeleteOutlined";
+import SaveIcon from "@mui/icons-material/Save";
+import CancelIcon from "@mui/icons-material/Close";
+import ArrowCircleDownIcon from "@mui/icons-material/ArrowCircleDown";
+import ArrowCircleUpIcon from "@mui/icons-material/ArrowCircleUp";
+import {
+    dbPackingSlotToInsert,
+    dbPackingSlotToDelete,
+    fetchPackingSlots,
+    swapRows,
+    dbPackingSlotToUpdate,
+} from "@/app/admin/packingSlotsTable/PackingSlotActions";
+import { LinearProgress } from "@mui/material";
+import { logError, logInfo } from "@/logger/logger";
+import { DatabaseError } from "@/app/errorClasses";
+
+interface EditToolbarProps {
+    setRows: (newRows: (oldRows: GridRowsProp) => GridRowsProp) => void;
+    setRowModesModel: (newModel: (oldModel: GridRowModesModel) => GridRowModesModel) => void;
+    rows: PackingSlotRow[];
+}
+
+export interface PackingSlotRow {
+    id: string;
+    name: string;
+    isShown: boolean;
+    order: number;
+    isNew: boolean;
+}
+
+function EditToolbar(props: EditToolbarProps): React.JSX.Element {
+    const { setRows, setRowModesModel, rows } = props;
+
+    const handleClick = (): void => {
+        const id = rows.length + 1;
+        setRows((oldRows) => [
+            ...oldRows,
+            { id, name: "", isShown: false, order: id, isNew: true },
+        ]);
+        setRowModesModel((oldModel) => ({
+            ...oldModel,
+            [id]: { mode: GridRowModes.Edit, fieldToFocus: "name" },
+        }));
+    };
+
+    return (
+        <GridToolbarContainer>
+            <Button color="primary" startIcon={<AddIcon />} onClick={handleClick}>
+                Add new slot
+            </Button>
+        </GridToolbarContainer>
+    );
+}
+
+const PackingSlotsTable: React.FC = () => {
+    const [rows, setRows] = useState<PackingSlotRow[]>([]);
+    const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({});
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+
+    useEffect(() => {
+        fetchPackingSlots()
+            .then((response) => setRows(response))
+            .catch((error) => {
+                const meta = {
+                    error: error,
+                };
+                void logError("Error fetching packing slots", meta);
```

Comment:
In javascript, `{ a: a }` can be simplified to `{ a }`
```suggestion
                void logError("Error fetching packing slots", { error });
```

Reply:
I think I wanted to handle this in the error handling ticket - haven't thought it through properly yet